pkg/operator/options/suite_test.go

```diff
@@ -0,0 +1,89 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "knative.dev/pkg/logging/testing"
+
+	"github.com/aws/karpenter-core/pkg/apis/settings"
+	"github.com/aws/karpenter-core/pkg/operator/options"
+)
+
+var ctx context.Context
+
+func TestSettings(t *testing.T) {
+	ctx = TestContextWithLogger(t)
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Settings")
+}
+
+var _ = Describe("Options", func() {
+	Context("Parse", func() {
+		DescribeTable(
+			"should successfully parse feature gates",
+			func(flagStr string, expectFunc func(*options.Options)) {
+				opts := options.New().MustParse("--feature-gates", flagStr)
+				expectFunc(opts)
+			},
+			Entry("Drift = true", "Drift=true", func(o *options.Options) {
+				Expect(o.FeatureGates.Drift).To(BeTrue())
+			}),
+			Entry("Drift = false", "Drift=false", func(o *options.Options) {
+				Expect(o.FeatureGates.Drift).To(BeFalse())
+			}),
+		)
+	})
+
+	Context("Merge", func() {
```

Comment:
Should we have some testing for defaults with parsing?

Reply:
Seems reasonable 👍